File: exercism/go/triangle/triangle.go
// Package triangle contains triangle utils
package triangle

import "math"

// Kind : type of Triangle
type Kind int

const (
	// NaT : not a triangle
	NaT = iota
	// Equ : equilateral
	Equ
	// Iso : isosceles
	Iso
	// Sca : scalene
	Sca
)

// KindFromSides returns type of triangle based on Kind
func KindFromSides(a, b, c float64) Kind {
	var k Kind

	switch {
	// For a shape to be a triangle at all,
	// all sides have to be of length > 0, and the
	// sum of the lengths of any two sides must be greater than or equal
	// to the length of the third side. See Triangle Inequality.
	case math.IsNaN(a) || math.IsNaN(b) || math.IsNaN(c):
	case math.IsInf(a, 1) || math.IsInf(b, 1) || math.IsInf(c, 1):
	case math.IsInf(a, -1) || math.IsInf(b, -1) || math.IsInf(c, -1):
	case a <= 0 || b <= 0 || c <= 0:
	case a+b < c || a+c < b || b+c < a:
		k = NaT
	// 	An equilateral triangle has all three sides the same length.
	case a == b && b == c:
		k = Equ
	// An isosceles triangle has at least two sides the same length. (It is sometimes specified as having exactly two sides the same length, but for the purposes of this exercise we'll say at least two.)
	case a == b || b == c || a == c:
		k = Iso
	// A scalene triangle has all sides of different lengths.
	default:
		k = Sca
	}

	return k
}

File: exercism/python/triangle/triangle.py
def is_triangle(sides):
    if len(sides) != 3:
        return False
    for x in range(0,3):
        if sides[x] <= 0:
            return False
        y = x+ 1
        if y == 3:
            y = 0
        z = y+1
        if z == 3:
            z = 0
        if sides[x] + sides[y] < sides[z]:
            return False
    return True

def get_equals_sides(sides):
    # print("sides=",sides)
    equals_sides = 0
    for x in range(0,3):
        y = 0 if x == 2 else x + 1
        # print( sides[x],sides[y])
        if sides[x] == sides[y]:
            equals_sides += 1
    return equals_sides

def equilateral(sides):
    return get_equals_sides(sides) == 3 and is_triangle(sides)

def isosceles(sides):
    return  get_equals_sides(sides) >= 1 and is_triangle(sides)

def scalene(sides):
    return sides[0] != sides[1] and sides[1] != sides[2] and sides[0] != sides[2] and is_triangle(sides)
